src/list.c

```diff
@@ -1520,15 +1520,13 @@ f_join(typval_T *argvars, typval_T *rettv)
 
     rettv->v_type = VAR_STRING;
 
-    if (in_vim9script()
```

Comment:
we don't need the `in_vim9script()` test anymore?

Reply:
The list type is checked for the regular Vim script just after this if statement.  As there is no need to check the list type twice, I removed the check for vim9script.